### PR TITLE
strip .gz from satellite files

### PR DESCRIPTION
**Pull request by ohnorobo**
This is a fix for an error I introduced in https://github.com/censoredplanet/censoredplanet-analysis/pull/101

Runtime error visible in https://console.cloud.google.com/errors/CJzcht_zxtaPTg?project=firehook-censoredplanet&time=P30D

### CODE REVIEW

File: pipeline/metadata/flatten_satellite.py [hunk 1]
```diff
@@ -191,72 +191,69 @@ def __init__(self, blockpage_matcher: BlockpageMatcher,
...
+
+    filename = pathlib.PurePosixPath(filepath).name
+    if '.gz' in pathlib.PurePosixPath(filename).suffixes:
+      filename = pathlib.PurePosixPath(filename).stem
+
+    if filename == SATELLITE_BLOCKPAGES_FILE:
+      yield from self._process_satellite_blockpages(scan, filepath)
     elif datetime.date.fromisoformat(date) < SATELLITE_V2_1_START_DATE:
-      yield from self._process_satellite_v1(date, scan, filename,
+      yield from self._process_satellite_v1(date, scan, filepath,
```
Comment (avirkud) [inline pipeline/metadata/flatten_satellite.py:215]: I think you could pass `filename` here. Then you don't need to duplicate the three lines in `_process_satellite_v1` (240-242).

File: pipeline/metadata/flatten_satellite.py [hunk 2]
```diff
@@ -191,72 +191,69 @@ def __init__(self, blockpage_matcher: BlockpageMatcher,
...
-    if pathlib.PurePosixPath(filename).name == SATELLITE_BLOCKPAGES_FILE:
-      yield from self._process_satellite_blockpages(scan, filename)
+    date = re.findall(r'\d\d\d\d-\d\d-\d\d', filepath)[0]
+
+    filename = pathlib.PurePosixPath(filepath).name
+    if '.gz' in pathlib.PurePosixPath(filename).suffixes:
+      filename = pathlib.PurePosixPath(filename).stem
+
+    if filename == SATELLITE_BLOCKPAGES_FILE:
+      yield from self._process_satellite_blockpages(scan, filepath)
```
Comment (avirkud) [inline pipeline/metadata/flatten_satellite.py:213]: The docstring for `_process_satellite_blockpages` needs to be updated (also the parameter name -> filepath?).

**Review by avirkud — CHANGES_REQUESTED** [review]

File: pipeline/metadata/flatten_satellite.py (hunk 1 above)
Comment (ohnorobo) [reply pipeline/metadata/flatten_satellite.py:215]: I'd rather keep passing the filepath and do the strip twice. 

I think we're going to want the filepath in the future. (For example to calculate a `source` field for the satellite table.)

File: pipeline/metadata/flatten_satellite.py (hunk 2 above)
Comment (ohnorobo) [reply pipeline/metadata/flatten_satellite.py:213]: updated

**Review by avirkud — APPROVED** [review]

**Pull request merged**